### ISSUE TITLE
RemoveTag for EC2 only removes the first item of tags: array

### ISSUE THREAD

**Issue by jtroberts83** [capitalone/cloud-custodian#2048]
https://github.com/capitalone/cloud-custodian/blob/master/c7n/tags.py#L389
Using the EC2 Resource if I try and run the following it only removes 'TagsNeedChecked' tag:
```
    - type: untag
      tags: ['TagsNeedChecked','EnvironmentTagChecked','ResourceContactTagChecked']
```
However if I run the following all 3 tags are removed:
```
    - type: untag
      tags: ['TagsNeedChecked']
    - type: untag
      tags: ['EnvironmentTagChecked']
    - type: untag
      tags: ['ResourceContactTagChecked']
  ```

**Comment by LykinsN** [comment]
I'll look into working this item.

**Comment by LykinsN** [comment]
I've opened a PR to fix this under #2062.

**Comment by jtroberts83** [comment]
Closing as linked to  #2062
